### PR TITLE
remove viper, use envconfig

### PR DESCRIPTION
**Pull request by skudasov**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -179,63 +120,3 @@ func (l *LocalStore) Fetch() ([]string, error) {
 	}
 	return l.RawKeys, nil
 }
-
-// ReadWriteRemoteRunnerConfig looks for an already existing remote config to read from, or asks the user to build one
-func ReadWriteRemoteRunnerConfig() (*RemoteRunnerConfig, error) {
```
Comment (skudasov) [inline config/config.go:184]: Should we write an example one for usage and place it with `framework.yaml` and `networks.yaml`?

File: config/config_models.go [hunk 2]
```diff
@@ -1,58 +1,106 @@
...
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+)
+
+// Config general framework config
+type Config struct {
+	FrameworkConfig    *FrameworkConfig    `envconfig:"FRAMEWORK_CONFIG_FILE" default:"../framework.yaml"`
+	NetworksConfig     *NetworksConfig     `envconfig:"NETWORKS_CONFIG_FILE" default:"../networks.yaml"`
```
Comment (skudasov) [inline config/config_models.go:16]: I propose the default path for any config to be `suite` or `e2e` directory, it's how it is in most of our repos

File: suite/soak/soak_runner_test.go [hunk 3]
```diff
@@ -7,34 +7,43 @@ import (
...
+func init() {
 	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
+}
+
+func TestSoak(t *testing.T) {
+	actions.LoadConfigs()
+	config.ProjectConfig.RemoteRunnerConfig = &config.RemoteRunnerConfig{
+		TestRegex:     "@soak-ocr",
+		TestDirectory: filepath.Join(utils.ProjectRoot, "./suite/soak/tests"),
+		SlackAPIKey:   "abcdefg",
```
Comment (kalverra) [inline suite/soak/soak_runner_test.go:29]: People will most definitely forget and check-in their Slack creds this way.

File: actions/actions.go [hunk 4]
```diff
@@ -40,47 +38,18 @@ const (
 var ContractDeploymentInterval = 500
 
 // GinkgoSuite provides the default setup for running a Ginkgo test suite
-func GinkgoSuite(frameworkConfigFileLocation string) {
-	LoadConfigs(frameworkConfigFileLocation)
+func GinkgoSuite() {
```
[the remark this answers is not in the document]
Comment (skudasov) [reply actions/actions.go:41]: Yes, `NETWORK_CONFIG_FILE="..."` etc, all paths get resolved as absolute ones

File: config/config.go (hunk 1 above)
Comment (skudasov) [reply config/config.go:184]: Added another readme, you can use both, YAML and env vars

**Review by kalverra — APPROVED** [review]

File: suite/soak/ocr_test_deploy.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+var _ = Describe("OCR Soak Setup @setup-soak", func() {
+	It("Deploys soak test runner", func() {
+		actions.LoadConfigs()
+		exePath := buildGoTests()
+
+		env, err := environment.DeployRemoteRunnerEnvironment(
+			environment.NewChainlinkConfig(
+				environment.ChainlinkReplicas(6, config.ChainlinkVals()),
+				"zchainlink-soak",
```
Comment (tateexon) [inline suite/soak/ocr_test_deploy.go:25]: Do you still want this named "zchainlink-soak", maybe something like "chainlink-ocr-soak"?

File: config/config_test.go [hunk 6]
```diff
@@ -80,22 +82,22 @@ func TestChainlinkValues(t *testing.T) {
...
-	_, err := config.ReadWriteRemoteRunnerConfig()
-	require.Error(t, err, "Wrote an example config file at %s. Please fill in values and log back in", utils.RemoteRunnerConfigLocation)
-	require.FileExists(t, utils.RemoteRunnerConfigLocation)
-
-	remoteConfig, err := config.ReadWriteRemoteRunnerConfig()
-	require.NoError(t, err)
-	require.Equal(t, "@soak-ocr", remoteConfig.TestRegex)
-	require.Equal(t, "abcdefg", remoteConfig.SlackAPIKey)
-}
+//func TestRemoteRunnerConfig(t *testing.T) {
```
Comment (tateexon) [inline config/config_test.go:85]: Do we still need this or can we toss it?

**Review by tateexon — APPROVED** [review]

File: suite/soak/ocr_test_deploy.go (hunk 5 above)
Comment (kalverra) [reply suite/soak/ocr_test_deploy.go:25]: I forgot to flag these, but I agree with both things Tate said.

**Pull request merged**